sale_order_warehouse_location/models/sale_order.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("partner_shipping_id")
+    def _compute_warehouse_id(self):
+        ret_vals = super()._compute_warehouse_id()
+        orders = self.filtered(
```

Comment:
You are iterating the same recordset twice which can decrease performance.
Instead of using `filtered` here you can check the check this condition in the loop below. Eg:

```python
for order in self:
    if not (o.partner_shipping_id and (o.state in ["draft", "sent"] or not o.ids)):
        continue
```